src/main/java/com/javastudy/coworkings/dao/jdbc/JdbcCoworkingDao.java

```diff
@@ -37,4 +43,31 @@ public Coworking getById(long id) {
             throw new RuntimeException("Coworking with id: " + id + "isn't found", e);
         }
     }
+
+    @Override
+    public List<Coworking> getTopEight() {
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement statement = connection.prepareStatement(GET_TOP_EIGHT)) {
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (!resultSet.next()) {
+                    throw new RuntimeException("Error happened while getting eight top-rated Coworkings");
+                }
+
+                List<Coworking> topEight = new ArrayList<>();
+                int i = 0;
+
+                while (resultSet.next()) {
+                    Coworking coworking = coworkingRowMapper.rowMap(resultSet);
+                    topEight.add(coworking);
+                    i++;
+                    if (i >= 8) {
+                        throw new RuntimeException("Found more than eight results of top-rated Coworkings, error happened;");
+                    }
```

Comment:
Do you really need this check? Do you know any case when GET_TOP_EIGHT select could return more than 8 rows?